src/components/Personalization/in-app-message-actions/actions/displayIframeContent.js

```diff
@@ -292,18 +286,18 @@ export default (settings, collect) => {
   return new Promise(resolve => {
     const { meta } = settings;
     displayHTMLContentInIframe(settings, (action, propositionAction) => {
-      const propositionEventTypes = new Set();
-      propositionEventTypes.add(PropositionEventType.INTERACT);
+      const propositionEventTypes = {};
+      propositionEventTypes[PropositionEventType.INTERACT] = EVENT_TYPE_TRUE;
 
       if (Object.values(PropositionEventType).indexOf(action) !== -1) {
-        propositionEventTypes.add(action);
+        propositionEventTypes[action] = EVENT_TYPE_TRUE;
       }
 
       collect({
         decisionsMeta: [meta],
         propositionAction,
         eventType: INTERACT,
-        propositionEventTypes: toArray(propositionEventTypes)
+        propositionEventTypes: Object.keys(propositionEventTypes)
```

Comment:
we can't use `toArray` here.  it does not work with Sets.  so i switched to another convention.